autometa/taxonomy/gtdb.py

```diff
@@ -0,0 +1,368 @@
+"""
+# License: GNU Affero General Public License v3 or later
+# A copy of GNU AGPL v3 should have been included in this software package in LICENSE.txt.
+
+File containing definition of the GTDB class and containing functions useful for handling GTDB taxonomy databases
+"""
+
+import gzip
+import logging
+import os
+import re
+import string
+import tarfile
+import glob
+from pathlib import Path
+
+from typing import Dict, List, Set, Tuple
+from itertools import chain
+from tqdm import tqdm
+from typing import Dict
+
+import pandas as pd
+import multiprocessing as mp
+
+from autometa.common.utilities import file_length, is_gz_file
+from autometa.common.external import diamond
+from autometa.taxonomy.database import TaxonomyDatabase
+from autometa.common.exceptions import DatabaseOutOfSyncError
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_gtdb_db(reps_faa: str, dbdir: str) -> str:
+    """
+    Generate a combined faa file to create the GTDB-t database.
+
+    Parameters
+    ----------
+    reps_faa : str
+        Directory having faa file of all representative genomes. Can be tarballed.
+    dbdir : str
+        Path to output directory.
+
+    Returns
+    -------
+    str
+        Path to combined faa file. This can be used to make a diamond database.
+    """
+
+    if reps_faa.endswith(".tar.gz"):
+        logger.debug(
+            f"Extracting tarball containing GTDB ref genome animo acid data sequences to: {dbdir}/protein_faa_reps"
+        )
+        tar = tarfile.open(reps_faa)
+        tar.extractall(path=dbdir)
+        tar.close()
+        logger.debug("Extraction done.")
+        reps_faa = dbdir
+
+    genome_protein_faa_filepaths = glob.glob(
+        os.path.join(reps_faa, "**", "*_protein.faa"), recursive=True
+    )
+
+    faa_index: Dict[str, str] = {}
+    for genome_protein_faa_filepath in genome_protein_faa_filepaths:
+        # Regex to get the genome accession from the file path
+        genome_acc_search = re.search(
+            r"GCA_\d+.\d?|GCF_\d+.\d?", genome_protein_faa_filepath
+        )
+        if not genome_acc_search:
+            continue
+        faa_index[genome_protein_faa_filepath] = genome_acc_search.group()
```

Comment:
This can be simplified by switching the boolean expression.

> In `re.search(pattern, file_string)` is the entire path returned from `genome_acc_search.group()`? or only the GCA or GCF string?

> Yep, looks like it simply returns the match

```python
genome_protein_faa_filepath = "gtdb_release207_v2_genomic_files_reps/genomic_files_reps/protein_faa_reps/bacteria/GB_GCA_018826885.1_protein.faa"

re.search(r"GCA_\d+.\d?|GCF_\d+.\d?", genome_protein_faa_filepath).group()
>>>'GCA_018826885.1'
```

```suggestion
        if genome_acc_search:
            faa_index[genome_protein_faa_filepath] = genome_acc_search.group()
```